parser_library/src/processing/statement_analyzers/hit_count_analyzer.cpp

```diff
@@ -29,148 +30,170 @@ hit_count_analyzer::hit_count_analyzer(context::hlasm_context& ctx)
-        }
+    return std::nullopt;
+};
+
+std::optional<stmt_lines_range> get_stmt_lines_range(const semantics::core_statement& statement,
+    statement_provider_kind prov_kind,
+    processing_kind proc_kind,
+    const context::hlasm_context& ctx)
+{
+    const auto& r = statement.stmt_range_ref();
 
-        return stmt_lines_compacter(r);
-    };
+    if (proc_kind != processing_kind::LOOKAHEAD && r.start == r.end)
+        return std::nullopt;
 
-    if (stmt->kind == context::statement_kind::RESOLVED)
+    if (prov_kind == statement_provider_kind::OPEN)
     {
-        if (auto resolved = stmt->access_resolved(); resolved)
-            return adj_range(resolved->stmt_range_ref());
-    }
-    else if (stmt->kind == context::statement_kind::DEFERRED)
-    {
-        if (auto deferred = stmt->access_deferred(); deferred)
-            return adj_range(deferred->stmt_range_ref());
+        const auto& source = ctx.current_source();
+        // TODO: look into the opencode range discrepancy
+        return std::make_pair(r.start.line, r.end.line + source.end_index - source.begin_index - 1);
     }
 
-    return std::nullopt;
+    return stmt_lines_compacter(r);
 }
 
-void emplace_hit_count(hit_count_map& hc_map,
-    const utils::resource::resource_location& rl,
-    stmt_lines_range lines_range,
-    bool can_be_external_macro,
-    bool increase_hit_count)
+void update_hc_details(
+    hit_count_details& hc_details, stmt_lines_range lines_range, bool can_be_external_macro, bool increase_hit_count)
 {
-    auto& start_line = lines_range.first;
-    auto& end_line = lines_range.second;
-
-    auto& stmt_hc_details = hc_map.try_emplace(rl).first->second;
-    stmt_hc_details.is_external_macro &= can_be_external_macro;
-    stmt_hc_details.max_lineno = std::max(stmt_hc_details.max_lineno, start_line);
+    const auto& [start_line, end_line] = lines_range;
 
-    auto& line_hits = stmt_hc_details.line_hits;
-    while (line_hits.size() <= end_line)
-        line_hits.resize(2 * line_hits.size());
+    hc_details.is_external_macro &= can_be_external_macro;
+    hc_details.max_lineno = std::max(hc_details.max_lineno, start_line);
 
-    std::for_each_n(
-        std::next(line_hits.begin(), start_line), end_line - start_line + 1, [increase_hit_count](auto& line_hit) {
-            line_hit.contains_statement = true;
+    auto& line_hits = hc_details.line_hits;
+    if (line_hits.size() <= end_line)
+        line_hits.resize(2 * end_line + 1000);
 
-            if (increase_hit_count)
-                line_hit.count++;
-        });
+    for (auto i = start_line; i <= end_line; ++i)
+    {
+        auto& line_hit = line_hits[i];
+        line_hit.contains_statement = true;
+        if (increase_hit_count)
+            line_hit.count++;
+    }
 }
 } // namespace
 
+hit_count_details& hit_count_analyzer::get_hc_details_reference(utils::resource::resource_location rl)
+{
+    return m_hit_counts.try_emplace(std::move(rl)).first->second;
+}
```

Comment:
```suggestion
hit_count_details& hit_count_analyzer::get_hc_details_reference(const utils::resource::resource_location& rl)
{
    return m_hit_counts.try_emplace(rl).first->second;
}
```
I think that not copying will be the common case.